### PR TITLE
Bitcoin gold

### PR DESCRIPTION
**Pull request by Vutov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "bitcoinjs-lib",
+  "name": "bitcoinforksjs-lib",
```
Comment (h4x3rotab) [inline package.json]: Let's rename it as `bitcoingoldjs-lib`

File: src/transaction.js [hunk 2]
```diff
@@ -33,8 +33,11 @@ Transaction.SIGHASH_ALL = 0x01
 Transaction.SIGHASH_NONE = 0x02
 Transaction.SIGHASH_SINGLE = 0x03
 Transaction.SIGHASH_ANYONECANPAY = 0x80
+Transaction.SIGHASH_BITCOINCASHBIP143 = 0x40
```
Comment (h4x3rotab) [inline src/transaction.js]: Let's rename it to `Transaction.SIGHASH_FORKID `

File: src/networks.js [hunk 3]
```diff
@@ -2,6 +2,28 @@
 // Dogecoin BIP32 is a proposed standard: https://bitcointalk.org/index.php?topic=409731
 
 module.exports = {
+  bitcoingold: {
+    messagePrefix: '\x18Bitcoin Gold Signed Message:\n',
```
Comment (h4x3rotab) [inline src/networks.js]: `\x18` is the hex of 24, which is the compact encoding of `len("Bitcoin Signed Message:\n")`.

As we are using "Bitcoin Gold Signed Message:\n", it should be `0x1d` (29 characters).

File: src/networks.js [hunk 4]
```diff
@@ -2,6 +2,28 @@
...
+    bip32: {
+      public: 0x0488b21e,
+      private: 0x0488ade4
+    },
+    pubKeyHash: 0x26,
+    scriptHash: 0x17,
+    wif: 0x80
+  },
+  bitcoingoldtestnet: {
+    messagePrefix: '\x18Bitcoin Gold Signed Message:\n',
```
Comment (h4x3rotab) [inline src/networks.js]: ditto

File: src/transaction.js [hunk 5]
```diff
@@ -402,6 +405,58 @@ Transaction.prototype.hashForWitnessV0 = function (inIndex, prevOutScript, value
...
+  // therefore we can pull out the setting of nForkHashType
+  // and pass it into the functions.
+
+  var nForkHashType = hashType
+  var fUseForkId = (hashType & Transaction.SIGHASH_BITCOINCASHBIP143) > 0
+  if (fUseForkId) {
+    nForkHashType |= Transaction.FORKID_BTG << 8
+  }
+
+  // BIP143 sighash activated in BitcoinCash via 0x40 bit
```
Comment (h4x3rotab) [inline src/transaction.js]: s/BitcoinCash/Bitcoin Gold

File: src/transaction.js [hunk 6]
```diff
@@ -402,6 +405,58 @@ Transaction.prototype.hashForWitnessV0 = function (inIndex, prevOutScript, value
...
+  if (sigVersion || fUseForkId) {
+    if (types.Null(inAmount)) {
+      throw new Error('Bitcoin Cash sighash requires value of input to be signed.')
```
Comment (h4x3rotab) [inline src/transaction.js]: s/Cash/Gold

**Review by h4x3rotab — CHANGES_REQUESTED** [review]
Good job! Please take a look at some small problems.

**Comment by qasimsaleem1** [comment]
Are we working on some marketing of btg and when new exchange will add btg like bitflyer

**Pull request merged**